model/task/task.go

```diff
@@ -2484,15 +2490,52 @@ func GetTasksByVersion(versionID, sortBy string, statuses []string, variant stri
 		countPipeline = append(countPipeline, stage)
 	}
 	countPipeline = append(countPipeline, bson.M{"$count": "count"})
-	sorters := bson.D{}
+
 	if len(sortBy) > 0 {
-		sorters = append(sorters, bson.E{Key: sortBy, Value: sortDir})
+		if sortBy == DisplayStatusKey {
+			// setting field `first` onto tasks with a failed status allows us to sort all failed statuses to top of query and then sort alphabetically
+			pipeline = append(pipeline, bson.M{
+				"$addFields": bson.M{
+					"first": bson.M{
+						"$cond": bson.M{
+							"if": bson.M{
+								"$or": bson.A{
+									bson.M{"$eq": bson.A{"$" + DisplayStatusKey, evergreen.TaskFailed}},
```

Comment:
I can imagine a function that takes as its single argument the list of failed task statuses and then returns this list. This would let someone modify the task failure statuses in globals.go without also having to remember to change this pipeline.

Reply:
great suggestion